libraries/Microsoft.Bot.Builder.Azure/CosmosDBKeyEscape.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Bot.Builder.Azure
+{
+    public static class CosmosDBKeyEscape
+    {
+        // The list of illegal characters for CosmosDB Keys comes from this list on
+        // the CosmostDB docs: https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.resource.id?view=azure-dotnet#remarks
+        private static char[] _illegalKeys = new[] { '\\', '?', '/', '#' };
+
+        // We are escaping illegal characters using a "*{AsciiCodeInHex}" pattern. This
+        // means a key of "?test?" would be escaped as "*3ftest*3f".
```

Comment:
so what happens if you are given a key such as `*3ftest*3f` don't you escape it to `*3ftest*3f` i.e. the same thing?
in which case you can get key conflict between `?test?` and `*3ftest*3f`
in other words don't you also need to escape the escape? or something like that.